src/engine/system/player_controlled.cpp

```diff
@@ -0,0 +1,31 @@
+#include "system.hpp"
+#include "engine/input/input_state.hpp"
+#include "engine/ecs.hpp"
+
+class SystemPlayerControlled: public System {
+  public:
+    void handle_components(ECS* ecs, InputState* input_state ) {
+      for (u32 ii=0; ii < ecs->comp_game_entity.size(); ii++) {
+        for (u32 jj=0; jj < ecs->comp_player_controlled.size(); ii++) {
+          if (ecs->comp_player_controlled[jj].entity_id == ecs->comp_game_entity[ii].entity_id) {
+            if (input_state->move_up >= 0) {
+              ecs->comp_game_entity[ii].acc.y = ecs->comp_game_entity[ii].acc.y - ecs->comp_player_controlled[jj].force_to_apply/ecs->comp_game_entity[ii].mass * input_state->move_up;
```

Comment:
Maybe clean up this function to make it more readable, have some local variables at the start of the loop with shorter names, like `Vec2* acc` and `f32 f`?